solver/progress.go

```diff
@@ -15,10 +15,20 @@ import (
 	"golang.org/x/sync/errgroup"
 )
 
+type Console interface {
```

Comment:
Maybe it would be good to have a comment that this is intended to match the `File` interface from `github.com/containerd/console`, to avoid pulling in that dependency. If that interface ever changes, we'll have to keep this copy in sync, and it will be easier to understand what happened with the comment giving context.

Reply:
Well, it is an interface that matches `*os.File`, just that `console` implemented that interface.  Not sure we care about console as much as matching `*os.File`?

Reply:
It needs to match what `progress.NewPrinter` expects as an argument.